src/acuvity/apexextend.py

```diff
@@ -35,22 +55,54 @@ def list_analyzer_names(self, group: Optional[str] = None) -> List[str]:
         """
         if self._available_analyzers is None:
             self._available_analyzers = self.list_analyzers()
+
         return sorted([ a.id for a in self._available_analyzers if (group is None or a.group == group) and a.id is not None ])
 
+        # # Filter out None values and sort
+        # return sorted([name for name in mapped_names if name is not None])
+
+    def list_detectable_secrets(self) -> list[str]:
+        """
+        list_secrets: returns a list of all available secrets that can be detected.
+        """
+        secrets_detector: list[str] = []
+        if self._available_analyzers is None:
+            self._available_analyzers = self.list_analyzers()
+        for analyzer in self._available_analyzers:
+            if analyzer.detectors:
+                secrets_detector = [
+                    str(detector.name)
+                    for detector in analyzer.detectors
+                    if detector.group == "Secrets"
+                ]
+                return secrets_detector
+        return secrets_detector
+
+    def list_detectable_pii(self) -> list[str]:
+        """
+        list_pii: returns a list of all available secrets that can be detected.
+        """
+        secrets_detector: list[str] = []
+        if self._available_analyzers is None:
+            self._available_analyzers = self.list_analyzers()
+        for analyzer in self._available_analyzers:
+            if analyzer.detectors:
+                secrets_detector = [
+                    str(detector.name)
+                    for detector in analyzer.detectors
+                    if detector.group == "PIIs"
+                ]
+                return secrets_detector
+        return secrets_detector
```

Comment:
- same comments as above
- additionally, please rename `secrets_detector` to something more suitable for this function here